### PR TITLE
Fix UART baudrate

### PR DESCRIPTION
**Pull request by ooxi**
The UART baudrate setup did not work as expected when using an 8 MHz clock. Setting the baudrate to 57600 would result in an actual baudrate of 9600.

In order to fix this behaviour, I'm changing the internal clock to 48 MHz using the [free-dap's samd21 system initialization routine](https://github.com/ataradov/free-dap/blob/master/platform/samd21/main.c#L53). This also significantly increases the read and write speed of SWD transfers.

### CODE REVIEW

**Comment by ooxi** [comment]
There seems to be some whitespace shenanigans with my patch, let me try again

**Comment by ataradov** [comment]
The point of this demo is to be simple, it is not a real project, just a demo. There is no need to make it more complicated than it needs to be.

I also see no reason why it would not be able to do 57600 at 8 MHz.

**Comment by ooxi** [comment]
> I also see no reason why it would not be able to do 57600 at 8 MHz.

I tried connecting to the virtual com port and only received garbage, when using the same baudrate on host and firmware. Only when changing to 9600 baud, the transfer could be read.

I did not try to make the demo more complicated, but simpler since it took me quite some time to figure out the baud rate for communication.

But most likely you are right and we should fix the uart communication when running 8 MHz instead of switching to 48 MHz

**Comment by ataradov** [comment]
This code is the same as in MCU starter projects. I just tried to switch that to 57600 and it works as expected.

**Comment by ooxi** [comment]
> This code is the same as in MCU starter projects. I just tried to switch that to 57600 and it works as expected.

Thanks for the investigation! I don't know, what I messed up.

**Comment by ooxi** [comment]
> Thanks for the investigation! I don't know, what I messed up.

For what it's worth: I copied the [samd21 platform code from free-dap](https://github.com/ataradov/free-dap/tree/master/platform/samd21) without changing the [`F_CPU` definition](https://github.com/ataradov/free-dap/blob/master/platform/samd21/make/Makefile#L45) thus giving the [`uart_init`](https://github.com/ataradov/embedded-swd/blob/master/main.c#L50) function a wrong indication of the correct CPU speed.
